perf/perf_fuzzer.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author:Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
+#
+
+import platform
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import archive, build, process, distro, genio
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Perffuzzer(Test):
+
+    """
+    This test run the perf fuzzer, which fuzz's the Linux perf_event
+    system call interface.
+    Source: https://github.com/deater/perf_event_tests/fuzzer
+    (http://web.eece.maine.edu/~vweaver/projects/perf_events/fuzzer/)
+    This test might crash the kernel
+    :avocado: tags=privileged
+    """
+
+    @staticmethod
+    def run_cmd_out(cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True, sudo=True)
+
+    def setUp(self):
+        '''
+        Install the packages
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in detected_distro.name:
+            kernel_ver = platform.uname()[2]
+            deps.extend(['linux-tools-common', 'linux-tools-%s'
+                         % kernel_ver])
+        # FIXME: "redhat" as the distro name for RHEL is deprecated
+        # on Avocado versions >= 50.0.  This is a temporary compatibility
+        # enabler for older runners, but should be removed soon
+        elif detected_distro.name in ['rhel', 'SuSE', 'fedora', 'redhat']:
+            deps.extend(['perf'])
+        else:
+            self.cancel("Install the package for perf supported by %s"
+                        % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def build_perf_test(self):
+        """
+        Building the perf event test suite
+        """
+        tarball = self.fetch_asset('perf-event.zip', locations=[
+                                   'https://github.com/deater/'
+                                   'perf_event_tests/archive/'
+                                   'master.zip'], expire='7d')
+        archive.extract(tarball, self.workdir)
+        self.sourcedir = os.path.join(self.workdir, 'perf_event_tests-master')
+        if build.make(self.sourcedir, extra_args="-s -S") > 0:
+            self.fail("Building perf event test suite failed")
```

Comment:
can this whole section moved to setUp sub routine   , as this section looks like test setup or at least from line 69 - line 74

Reply:
It would be clean to have system pre-requisite in setup and move all test related things to build function. Because as the function does nothing beyond setting up the testcase. It should be something like extended setup function. If you insist, we should move the testcase downloading to setup, I can re-spin the patch.

Reply:
@kamalesh-babulal no  hard feeling here , just i want to make uniform all the test cases in repository and it also aligned with purpose of setup

Reply:
Sure, I will address it